### PR TITLE
fix timeline ui flicking back and forth on live streams

### PR DESCRIPTION
**Pull request by Cagince**
moved playback position calculation logic inside a method and used each time for consistency.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,14 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/) 
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [develop]
+
+### Fixed
+- UI Flickering back and forth on live streams after AD's
+
+### Changed
+- Seekbar doesn't use `SmoothPositionUpdater` on Live streams anymore.
```
Comment (stonko1994) [inline CHANGELOG.md]: This entry is actually not true 🤔. The `startSmoothPlaybackPositionUpdater` was never used on Live streams. (It was a bug that it was). So please remove this entry.

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,6 +4,14 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](http://keepachangelog.com/) 
 and this project adheres to [Semantic Versioning](http://semver.org/).
 
+## [develop]
+
+### Fixed
+- UI Flickering back and forth on live streams after AD's
```
Comment (stonko1994) [inline CHANGELOG.md]:
```suggestion
- UI flickering back and forth on live streams after AD's
```

File: spec/components/seekbar.spec.ts [hunk 3]
```diff
@@ -1,6 +1,9 @@
 import { MockHelper, TestingPlayerAPI } from '../helper/MockHelper';
 import { SeekBar } from '../../src/ts/components/seekbar';
 import { UIInstanceManager } from '../../src/ts/uimanager';
+import { PlayerUtils } from '../../src/ts/playerutils';
```
Comment (stonko1994) [inline spec/components/seekbar.spec.ts]: Please remove unused imports.

File: CHANGELOG.md (hunk 2 above)
Comment (protyposis) [inline CHANGELOG.md]:
```suggestion
- UI Flickering back and forth on live streams after ads
```

File: spec/components/seekbar.spec.ts [hunk 4]
```diff
@@ -1,6 +1,9 @@
...
+import { anyTypeAnnotation } from '@babel/types';
```
Comment (protyposis) [inline spec/components/seekbar.spec.ts]: What is this?

File: src/ts/components/seekbar.ts [hunk 5]
```diff
@@ -161,16 +161,14 @@ export class SeekBar extends Component<SeekBarConfig> {
         if (player.getMaxTimeShift() === 0) {
           // This case must be explicitly handled to avoid division by zero
           this.setPlaybackPosition(100);
-        }
-        else {
+        } else {
```
Comment (protyposis) [inline src/ts/components/seekbar.ts]: unnecessary change

File: src/ts/components/seekbar.ts [hunk 6]
```diff
@@ -161,16 +161,14 @@ export class SeekBar extends Component<SeekBarConfig> {
...
           let playbackPositionPercentage = 100 - (100 / player.getMaxTimeShift() * player.getTimeShift());
           this.setPlaybackPosition(playbackPositionPercentage);
         }
 
         // Always show full buffer for live streams
         this.setBufferPosition(100);
-      }
-      else {
+      } else {
```
Comment (protyposis) [inline src/ts/components/seekbar.ts]: unnecessary change

File: src/ts/components/seekbar.ts [hunk 7]
```diff
@@ -360,10 +361,13 @@ export class SeekBar extends Component<SeekBarConfig> {
    * Update seekbar while a live stream with DVR window is paused.
    * The playback position stays still and the position indicator visually moves towards the back.
    */
-  private configureLivePausedTimeshiftUpdater(player: PlayerAPI, uimanager: UIInstanceManager,
-                                              playbackPositionHandler: () => void): void {
+  private configureLivePausedTimeshiftUpdater(
+    player: PlayerAPI,
+    uimanager: UIInstanceManager,
+    playbackPositionHandler: () => void,
+  ): void {
```
Comment (protyposis) [inline src/ts/components/seekbar.ts]: unnecessary change

File: src/ts/components/seekbar.ts (hunk 5 above)
Comment (Cagince) [reply src/ts/components/seekbar.ts]: these changes were mainly to refine the code, might be non-relevant for this PR but it's ok to keep consistency I guess

**Comment by Cagince** [comment]
@protyposis sorry about that, apparently I forgot to stash before changing the branches :/ 
thank you for the warning :+1:  won't happen again
